spec/classes/init_spec.rb

```diff
@@ -260,6 +260,48 @@
 
   end
 
+  context 'with selboolean specified as a hash' do
+    let(:facts) { { :osfamily => 'RedHat' } }
+    let :params do
+      {
+        :selboolean_hiera_merge => 'false',
+        :selboolean => {
+          'nfs_export_all_ro' => {
+            'value' => 'on',
+          },
+          'nfs_export_all_rw' => {
+            'persistent' => true,
+            'value'      => 'on',
+          },
+        }
+      }
+    end
+
+    it { should contain_class('types') }
+    it {
+      should contain_selboolean('nfs_export_all_ro').with({
+        'value' => 'on',
+      })
+    }
+    it {
+      should contain_selboolean('nfs_export_all_rw').with({
+        'persistent' => 'true',
```

Comment:
remove the quotes. You're saying it should be a string with the value true though the hash above is specifying the boolean true.